fedimint-core/src/encoding/tls.rs

```diff
@@ -10,3 +10,49 @@ impl Encodable for rustls::Certificate {
         self.0.encode_hex::<String>().consensus_encode(writer)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Cursor;
+
+    use super::*;
+
+    struct MockCertificate(Vec<u8>);
+
+    impl MockCertificate {
+        fn new(data: Vec<u8>) -> Self {
+            MockCertificate(data)
+        }
+    }
+
+    impl Encodable for MockCertificate {
+        fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+            self.0.encode_hex::<String>().consensus_encode(writer)
+        }
+    }
```

Comment:
1. This encoding is horribly broken (first encodes to hex, then as a byte vector)
2. It doesn't test any actual code from the module it is in, you just implement some mock that is never used elsewhere and test that (it *kinda* tests some of the core encoding functions, but such a test should live elsewhere and wouldn't require the `MockCertificate`)